src/sql/src/plan/statement/show.rs

```diff
@@ -474,8 +476,10 @@ pub fn show_indexes<'a>(
             );
         }
         query_filter.push(format!("on_id = '{}'", on_item.id()));
-    } else {
-        let schema_spec = scx.resolve_optional_schema(&from_schema)?;
+    }
+
+    if let Some(schema) = from_schema {
+        let schema_spec = schema.schema_spec();
```

Comment:
How do we know here that this won't panic? Because `show_indexes()` won't be called at all if name resolution failed?

Reply:
Yeah, the `Error` variant is supposed to always return an error during name resolution and never make it any further.